### DISCUSSION TITLE
Problem using struts 1.5 RC version with moduler struts configuration files

### DISCUSSION THREAD

**Discussion by psinghwatermark** [ste-gr/struts1-reloaded#27] [Q&A] [answered]
Hi Team,

I am testing struts [1.5.0-RC1] with my enterprise application, bumping from struts 1.1 to struts [1.5.0-RC1] due to jakarta change with tomcat 10.1.x but facing issue using multi module configuration for struts in action servlet init parameters.  Below is my configuration snippet from web.xml action servlet init params

```
<!-- Struts configuration -->
<init-param>
	<param-name>config</param-name>
	<param-value>/apps/myapp/struts-config-higher.xml</param-value>
</init-param>
<init-param>
	<param-name>config/common</param-name>
	<param-value>/WEB-INF/struts-config-common.xml</param-value>
</init-param>
<init-param>
	<param-name>config/home</param-name>
	<param-value>/WEB-INF/struts-config-home.xml</param-value>
</init-param>
```

Problem that i am facing is form-beans, actions, forwards definitions etc mentioned in other files except first config file are NOT available when executing struts action. Debugging results shows that struts init phase is successful in reading the modular configuration but component definitions are not available while execution. We were using older way of deceleration before with struts1.1 by separating file names with comma under single config param but it throws error using struts [1.5.0-RC1]. 
**Exception : "ValidatorPlugIn cannot be redefined for module ' ' "**

I am not sure if it is a bug or configuration issue. Please suggest correct way to pass multiple struts files with struts [1.5.0-RC1]. 

Please let me know in case any further details are required to understand it better.

**Comment by ste-gr** [comment, marked as answer]
Hi @psinghwatermark,

at first glance, your configuration in the `web.xml` looks fine. Have you also updated the DTDs in your configuration files?

May this [Example-App](../blob/main/apps/examples/src/main/webapp/WEB-INF/web.xml) can give you some hints.

Upgrading from 1.1 directly to 1.5 is a big step. I would try to make small steps and upgrade your application from 1.1 to 1.2 to 1.3 to 1.4 to 1.5. I think, all Struts versions up to 1.4 should run with Tomcat 9. The step from Struts 1.4 to 1.5 and Tomcat 9 to 10.1 is then only very small.

The Apache Struts team also provides upgrade notes:
[Old Struts-Wiki - Struts Upgrade](https://web.archive.org/web/20170711220110/https://wiki.apache.org/struts/StrutsUpgrade)

Maybe there are some notes there that will helps you with the first steps.

From 1.3.11 onwards, I have tried to document the breaking-changes in the release documentation: [Releases](../releases)

I hope I have been able to help you a little
Greetings
Stefan

**Reply by psinghwatermark** [reply]
Hi @ste-gr,
Thanks for pointing out exact steps for struts upgrade, i mistakenly mentioned struts 1.1 though we were on struts 1.3 already. I updated my project DTD's and configuration files as per upgrade notes, but multi moduler struts config is still not working. If i combine whole config to a single file, it works fine. But upgrade notes were a great help, Thanks!
psingh

**Reply by ste-gr** [reply]
Hi @psinghwatermark,

thanks for your reply. It is a pity that the multi module configuration has not (yet) worked. I'm sure it's just a minor issue.

Greetings
Stefan
